promises.cf

```diff
@@ -57,6 +57,10 @@ body common control
 
       version => "CFEngine Promises.cf 3.7.0";
 
+      cfengine_3_7.(debian||redhat)::
+          package_inventory => { $(package_module_knowledge.platform_default) };
+          package_module => $(package_module_knowledge.platform_default);
```

Comment:
Do these have sensible defaults inside of the C code? They could be hash commented out with a note saying the syntax is incompatible prior to 3.7. 

Splitting promises.cf is possible, but really only from execd where you could choose a different promsies.cf file. You would still end up out of luck for the actual promsies.cf file. I guess promises.cf would be the 3.6 compat version, and you would have to have a different one for 3.7. Very nasty. I'd rather see a commented block.